### DISCUSSION TITLE
USD export options, Bumps, Normals, Displacements and Catclak attributes

### DISCUSSION THREAD

**Discussion by JonnhyRDG** [Autodesk/maya-usd#1534] [Q&A]
Hi everyone!

I've been testing passing a whole project to Katana through the Maya´s USD export plugin. Only to find there are plenty of attributes that don't get exported correctly or at all.

Examples:

- Geometry attributes, catclack and iterations get ignored. There's a @"subdivision method" flag at the export window, which applies a catclark iteration of 1 to everything. **<-- ok I just read that this is a known function that hasn´t been implemented yet, correct me if I´m wrong.**

- If you only connect a single channel to a shader attribute, it gets ignore (typical use case, connect only the "r" channel to a transmission weight, or what have you - there's a workaround for this, which is to use the aiRgbaToFloat though).

- aiBump2d does not work at all? Issue with normals? Can't make it work at all.

- aiNormalMap attributes "color to signed" and "tangent space" won't get exported, and are ignored as defaults. Hence not getting the input from those options and no way to activate it.

I'm VERY eager to hear from anyone who got any of those exporting correctly, any input would be much appreciated.
Or if this is a current limitation it would also be very helpful to know, so I stop banging my head against the proverbial wall.

Thanks for reading.

**Specs:**
 - Windows 10
 - Maya 2022
 - Maya USD 0.10

**Comment by sebastienblor** [comment]
For others looking at this post, Jonnhy also asked the same question in arnold-answers and I've replied in there.
These are known limitations of the mayaUSD export with Arnold. For now, the only way to have all the arnold data in USD files is still to use the Arnold USD exports from the arnold menus. But we're currently working on improving this, so that in the future you can have all the arnold data in maya usd exports.

**Reply by kxl-adsk** [reply]
FWIW. I converted this to a discussion post. As well, here is the direct link to the arnold-answers question - https://answers.arnoldrenderer.com/questions/34390/usd-export-options-bumps-normals-displacements-and.html
